pkg/kapp/cmd/app/apply_flags.go

```diff
@@ -63,6 +63,8 @@ func (s *ApplyFlags) SetWithDefaults(prefix string, defaults ApplyFlags, cmd *co
 		5, "Maximum number of concurrent wait operations")
 
 	cmd.Flags().BoolVar(&s.ExitStatus, prefix+"apply-exit-status", false, "Return specific exit status based on number of changes")
+
+	cmd.Flags().BoolVar(&s.ExitOnError, prefix+"exit-on-apply-wait-error", false, "Exit quickly on apply or wait failure")
```

Comment:
Did we not wanna set the default to true to start off with? (as per the discussion in the community meeting, where more users would like to know immediately If something is not working out)

nit: "Exit quickly on" => "Exit immediately on"/"Exit on"
I guess quickly is a bit more open ended than immediately?

nit:(to self) I am alright with flag name but gonna give it some thought 🤔

Reply:
> Did we not wanna set the default to true to start off with?

Yeah, that's something I am still thinking about. I am okay with setting default to true for now.

> nit: "Exit quickly on" => "Exit immediately on"/"Exit on"

Hehe, I started with `Exit on...` at first but realised that we are always going to exit in case of a failure, but yeah `Exit immediately...` makes more sense.

Reply:
> I am still having second thoughts about the name, so open to some suggestions.

I cannot think of anything that is as easy to understand as this one.

> Yeah, that's something I am still thinking about. I am okay with setting default to true for now.

I think that might be the way to go judging from the discussion we had. Also prevents this from being a breaking change, even though it is not a mean one.

Reply:
Bumping this ^ 
I think this is the major change blocking this?

Reply:
Yep, will have a better answer once I have done some more testing with this enabled.
(Although I am thinking that we should go ahead with the existing behaviour i.e exit on error, so that we can get some feedback about how intuitive the output is before we go on and make this as a default behaviour)